### PR TITLE
Fix yet more random test failures on ILC

### PR DESCRIPTION
**Pull request by AtsushiKan**
- FileVersionInfo.Tests

  TestAssembly.dll needs to exist (in its IL form)
  in the test directory - Fix is to mangle the file
  extension so that ILC treats it as a data asset
  rather as IL to be merged into the executable.

- Other stuff is self-explanatory.

### CODE REVIEW

**Comment by stephentoub** [comment]
Why is the .ildll file getting checked in?  It can't be built as part of the tests?

File: src/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Misc.cs [hunk 1]
```diff
@@ -1068,6 +1068,9 @@ public void Dev10_40561()
             return;
         }
 
+        // Test failure on ILC: Test depends on Xml Serialization and requires reflection on a LOT of types under System.Xml.Schema namespace.
+        // Rd.xml with "<Namespace Name="System.Xml.Schema" Dynamic="Required Public" />" lets this test pass but we should probably be
+        // fixing up XmlSerializer's own rd.xml rather than the test here.
```
Comment (stephentoub) [inline src/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Misc.cs]: Is there an issue number associated with this?

Comment (AtsushiKan) [inline src/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Misc.cs]: Nope. The fact that the test will continue failing on ILC will keep it on our radar.

**Comment by AtsushiKan** [comment]
>Why is the .ildll file getting checked in?

1. It's consistent with the other PE assets used by this test being checked in.

2. Copying it after the fact still causes ILC to build TestAssembly.dll, which is incorrect and time-consuming.

3. MSBuild and I are mortal enemies. The simpler (and fewer) msbuild hacks there are, the better my life is.

**Comment by stephentoub** [comment]
> It's consistent with the other PE assets used by this test being checked in.

And it's not great that we've already done that 😦  We have a separate https://github.com/dotnet/corefx-testdata repo where we've generally checked in binary files.

> The simpler (and fewer) msbuild hacks there are, the better my life is.

It may make your life better, but adding binary files to git makes everyone's life a little bit harder. Once a file is in git's history, it basically will always be there, even if a subsequent commit deletes it, so from this point on every clone of the repo will pull down this 5K binary file whether it's still there or not.

**Pull request merged**